programs/voter-stake-registry/src/state/deposit_entry.rs

```diff
@@ -158,34 +204,59 @@ impl DepositEntry {
 
         // Other lockup types don't need changes, because the user
         // cannot reduce their lockup strength.
-
-        altered.voting_power_locked(at_ts, max_locked_vote_weight, lockup_saturation_secs)
+        altered.voting_power_locked(
+          at_ts, 
+          min_required_locked_vote_weight,
+          max_locked_vote_weight, 
+          lockup_saturation_secs,
+          min_required_lockup_saturation_secs
+        )
     }
 
     /// Vote power contribution from funds with linear vesting.
     fn voting_power_cliff(
         &self,
         curr_ts: i64,
+        min_required_locked_vote_weight: u64,
         max_locked_vote_weight: u64,
         lockup_saturation_secs: u64,
+        min_required_lockup_saturation_secs: u64,
     ) -> Result<u64> {
-        let remaining = min(self.lockup.seconds_left(curr_ts), lockup_saturation_secs);
-        Ok(u64::try_from(
-            (max_locked_vote_weight as u128)
-                .checked_mul(remaining as u128)
-                .unwrap()
-                .checked_div(lockup_saturation_secs as u128)
-                .unwrap(),
-        )
-        .unwrap())
+        let total_lockup_secs = self.lockup.total_seconds();
+
+        if total_lockup_secs <= min_required_lockup_saturation_secs {
+            let remaining = min(self.lockup.seconds_left(curr_ts), min_required_lockup_saturation_secs);
```

Comment:
I'm confused:

  1. Is total seconds less than min required lockup even ever allowed?
  2. How could seconds_left ever be greater than `min_required_lockup_saturation_secs` if the if statement above is true?

Reply:
No that <= should never happen, ill change the comparison to ==